src/components/dashboard/Dashboard.js

```diff
@@ -258,9 +260,10 @@ const Dashboard = props => {
 
   showOutOfGasError(props)
 
-  const handleWithdraw = async () => {
+  const handleWithdraw = async (attempts: number) => {
     const { paymentCode, reason } = props.navigation.state.params
     const { styles }: DashboardProps = props
+    const activeAttempts = attempts ? attempts : 0
```

Comment:
here use the getStatus method
if completed show dialog with already withdrawn
if pending executewithdraw
if unknown check status again in 2 seconds 3 times.